src/compass/utils/ale.py

```diff
@@ -0,0 +1,398 @@
+    if not os.path.exists(cslc_file):
+        err_str = f'{cslc_file} CSLC-S1 product does not exist'
+        raise FileNotFoundError(err_str)
+
+    # Check corner reflector file exists
+    if not os.path.exists(cr_file):
+        err_str = f'{cr_file} corner reflector file does not exist'
+        raise FileNotFoundError(err_str)
+
+    # Open corner reflector file
+    cr_df = pd.read_csv(cr_file)
+
+    # Add empty columns necessary for computation
+
+    # Identify which CR are contained in the usable part of the burst
+    # Get CSLC-S1 polygon
+    cslc_poly = get_cslc_polygon(cslc_file)
+    cr_x = []
+    cr_y = []
+    cr_x_cslc = []
+    cr_y_cslc = []
+
+    for idx, row in cr_df.iterrows():
+        cr_lat = row['Latitude (deg)']
+        cr_lon = row['Longitude (deg)']
+        cr_loc = geometry.Point(cr_lon, cr_lat)
+
+        if cslc_poly.contains(cr_loc):
+            # Convert corner lat/lon coordinates in UTM
+            cslc_epsg = get_cslc_epsg(cslc_file)
+
+            # Correct corner reflector position for solid Earth tides
+            # otherwise just transform coordinates to UTM
+            if correct_set:
+                x, y = correct_cr_tides(cslc_file, cr_lat, cr_lon)
+            else:
+                x, y = latlon2utm(cr_lat, cr_lon, cslc_epsg)
+
+            cr_x.append(x)
+            cr_y.append(y)
+
+            # Compute location of CR in CSLC image at pixel precision
+            x_coord_vect, dx, y_coord_vect, dy = get_xy_info(cslc_file)
+            cr_x_cslc.append(int((x - x_coord_vect[0]) / dx))
+            cr_y_cslc.append(int((y - y_coord_vect[0]) / dy))
+        else:
+            # If the CR is not in the burst, drop the corresponding
+            # row from the panda dataframe
+            cr_df.drop(idx, inplace=True)
+
+    # Assign computed data
+    cr_df['CR_X'] = cr_x
+    cr_df['CR_Y'] = cr_y
+    cr_df['CR_X_CSLC'] = cr_x_cslc
+    cr_df['CR_Y_CSLC'] = cr_y_cslc
+
+    x_peak_vect = []
+    y_peak_vect = []
+    
+    # Find peak location for every corner reflector in DataFrame
+    cslc_arr = get_cslc(cslc_file)
+    for idx, row in cr_df.iterrows():
+        x_peak, y_peak = find_peak(cslc_arr, int(row['CR_X_CSLC']),
+                                   int(row['CR_Y_CSLC']))
+        x_coord_vect, dx, y_coord_vect, dy = get_xy_info(cslc_file)
+        x_peak_vect.append(x_coord_vect[0] + x_peak * dx)
+        y_peak_vect.append(y_coord_vect[0] + y_peak * dy)
+
+    cr_df['CR_X_CSLC_PEAK'] = x_peak_vect
+    cr_df['CR_Y_CSLC_PEAK'] = y_peak_vect
+
+    # Compute absolute geolocation error along X and Y direction
+    cr_df['ALE_X'] = cr_df['CR_X_CSLC_PEAK'] - cr_df['CR_X']
+    cr_df['ALE_Y'] = cr_df['CR_Y_CSLC_PEAK'] - cr_df['CR_Y']
+
+    if csv_output_file is not None:
+        cr_df.to_csv(csv_output_file)
+
+    if plot_ale:
+        fig, ax = plt.subplots(figsize=(8, 6))
+        sc = ax.scatter(cr_df['ALE_Y'], cr_df['ALE_X'], s=200, alpha=0.8,
```

Comment:
X and Y are swapped?
```suggestion
        sc = ax.scatter(cr_df['ALE_X'], cr_df['ALE_Y'], s=200, alpha=0.8,
```

Reply:
Resolved, thanks.